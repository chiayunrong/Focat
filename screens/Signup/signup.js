import { KeyboardAvoidingView, StyleSheet, Text } from 'react-native';
import React, { useState } from 'react';
import { auth, firestore } from '../../firebase';
import { setDoc, doc } from 'firebase/firestore';
import { createUserWithEmailAndPassword } from 'firebase/auth';
import Toast from 'react-native-root-toast';
import { Button, Input, Layout, Icon } from '@ui-kitten/components';
import EmailTextInput from '../../components/email_textinput';
import PasswordTextInput from '../../components/password_textinput';
import { storeUsername } from '../../utils/usernameStorage';
import { default as theme } from '../../assets/custom-theme.json';

const Signup = () => {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [secureTextEntry, setSecureTextEntry] = useState(true);

  /**
   * Passwords strength must be strong, with at least 8 characters,
   * at least 1 upper and lower case letter, 1 digit, and 1 special character.
   */
  const validatePasswordStrength = () => {
    const regex = new RegExp(
      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&+])[A-Za-z\d@$!%*?&+]{8,}$/i
    );
    return regex.test(password);
  };

  /**
   * Function that will be executed when user presses "Signup".
   * All fields must be filled up.
   * Password and confirm password must match.
   * Strength of the password entered must be strong.
   * Email entered must be of valid format.
   */
  const handleSignup = () => {
    if (username === '') {
      Toast.show('Please fill up username.', {
        duration: Toast.durations.SHORT,
      });
    } else if (password !== confirmPassword) {
      Toast.show('Password and confirm password must be the same.', {
        duration: Toast.durations.SHORT,
      });
    } else if (!validatePasswordStrength()) {
      Toast.show(
        'Password should have a minimum of 8 characters, with at least a lower and upper case letter, digit, and special character.',
        {
          duration: Toast.durations.SHORT,
        }
      );
    } else {
      storeUsername(username);
      createUserWithEmailAndPassword(auth, email, password)
        .then((userCredential) => {
          const user = userCredential.user;
          setDoc(doc(firestore, 'users', user.uid), {
            username: username.trim(),
            cat_cash: 0,
            is_online: true,
            sent_requests: [],
            received_requests: [],
            friends: [],
            achievements: [],
            items: [],
            current_session: '',
          });
          console.log('Registered:', user.email);
          Toast.show(`Successful sign up. Welcome ${username}!`, {
            duration: Toast.durations.SHORT,
          });
        })
        .catch((error) => {
          if (error.code === 'auth/email-already-in-use') {
            Toast.show('Email already in use.', {
              duration: Toast.durations.SHORT,
            });
          } else if (error.code === 'auth/invalid-email') {
            Toast.show('Invalid email.', {
              duration: Toast.durations.SHORT,
            });
          } else {
            Toast.show('Unable to create an account. Try again.', {
              duration: Toast.durations.SHORT,
            });
          }
        });
    }
  };

  return (
    <Layout style={styles.container}>
      <KeyboardAvoidingView>
        <Text style={styles.text}>Enter the world of Focat.</Text>
        <Input
          style={styles.input}
          placeholder="Username"
          value={username}
          maxLength={15}
          onChangeText={setUsername}
          accessoryLeft={(props) => <Icon name="person-outline" {...props} />}
          testID={'usernameInput'}
          selectionColor={theme['color-primary-300']}
        />
        <EmailTextInput
          email={email}
          setEmail={setEmail}
          testID={'emailInput'}
        />
        <PasswordTextInput
          password={password}
          setPassword={setPassword}
          secureTextEntry={secureTextEntry}
          setSecureTextEntry={setSecureTextEntry}
          testID={'passwordInput'}
        />
        <Input
          style={styles.input}
          placeholder="Re-enter Password"
          value={confirmPassword}
          onChangeText={setConfirmPassword}
          secureTextEntry={secureTextEntry}
          accessoryLeft={(props) => <Icon name="lock-outline" {...props} />}
          testID={'confirmPasswordInput'}
          selectionColor={theme['color-primary-300']}
        />
        <Button style={styles.button} onPress={handleSignup}>
          Sign Up
        </Button>
      </KeyboardAvoidingView>
    </Layout>
  );
};

export default Signup;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    marginBottom: 64,
    fontSize: 24,
    textAlign: 'center',
    width: 300,
  },
  input: {
    marginVertical: 10,
  },
  button: {
    alignItems: 'center',
    marginVertical: 10,
  },
});
