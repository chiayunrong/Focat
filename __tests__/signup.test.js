import React from 'react';
import { create, act } from 'react-test-renderer';
import * as eva from '@eva-design/eva';
import { RootSiblingParent } from 'react-native-root-siblings';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import {
  ApplicationProvider,
  IconRegistry,
  Button,
} from '@ui-kitten/components';
import Signup from '../screens/Signup/signup';
jest.useFakeTimers();

const userObject = {
  username: 'Test User',
  email: 'testuser@email.com',
  password: 'Password123!',
};

const getComponent = () => {
  return create(
    <RootSiblingParent>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={eva.light}>
        <Signup />
      </ApplicationProvider>
    </RootSiblingParent>
  );
};

describe('Snapshot Test', () => {
  test('renders correctly', () => {
    const component = getComponent().toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Signup Function', () => {
  test('renders sign up button', () => {
    const component = getComponent();
    const signupButton = component.root.findByType(Button);
    expect(signupButton).toBeTruthy();
  });

  test('updates username input field', () => {
    const component = getComponent();
    const usernameInput = component.root.findByProps({
      testID: 'usernameInput',
    });
    act(() => {
      usernameInput.props.onChangeText(userObject.username);
    });
    expect(usernameInput.props.value).toBe(userObject.username);
  });

  test('updates email input field', () => {
    const component = getComponent();
    const emailInput = component.root.findByProps({ testID: 'emailInput' });
    act(() => {
      emailInput.props.setEmail(userObject.email);
    });
    expect(emailInput.props.email).toBe(userObject.email);
  });

  test('updates password input field', () => {
    const component = getComponent();
    const passwordInput = component.root.findByProps({
      testID: 'passwordInput',
    });
    act(() => {
      passwordInput.props.setPassword(userObject.password);
    });
    expect(passwordInput.props.password).toBe(userObject.password);
  });

  test('updates confirm password input field', () => {
    const component = getComponent();
    const passwordInput = component.root.findByProps({
      testID: 'confirmPasswordInput',
    });
    act(() => {
      passwordInput.props.onChangeText(userObject.password);
    });
    expect(passwordInput.props.value).toBe(userObject.password);
  });
});
