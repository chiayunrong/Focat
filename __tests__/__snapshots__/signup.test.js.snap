// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "marginBottom": 64,
            "textAlign": "center",
            "width": 300,
          }
        }
      >
        Enter the world of Focat.
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginVertical": 10,
            "opacity": 1,
          }
        }
        testID="usernameInput"
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#8F9BB3"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "marginHorizontal": 8,
                    "tintColor": "#8F9BB3",
                    "width": 24,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4287601587}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup>
                  <RNSVGGroup>
                    <RNSVGRect
                      height="24"
                      opacity={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M12 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4zm0-6a2 2 0 1 1-2 2 2 2 0 0 1 2-2z"
                    />
                    <RNSVGPath
                      d="M12 13a7 7 0 0 0-7 7 1 1 0 0 0 2 0 5 5 0 0 1 10 0 1 1 0 0 0 2 0 7 7 0 0 0-7-7z"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            appearance="default"
            editable={true}
            maxLength={15}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Username"
            placeholderTextColor="#8F9BB3"
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                undefined,
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginVertical": 10,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#8F9BB3"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "marginHorizontal": 8,
                    "tintColor": "#8F9BB3",
                    "width": 24,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4287601587}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup>
                  <RNSVGGroup>
                    <RNSVGRect
                      height="24"
                      opacity={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm-.67 2L12 10.75 5.67 6zM19 18H5a1 1 0 0 1-1-1V7.25l7.4 5.55a1 1 0 0 0 .6.2 1 1 0 0 0 .6-.2L20 7.25V17a1 1 0 0 1-1 1z"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Email"
            placeholderTextColor="#8F9BB3"
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                undefined,
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginVertical": 10,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#8F9BB3"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "marginHorizontal": 8,
                    "tintColor": "#8F9BB3",
                    "width": 24,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4287601587}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup>
                  <RNSVGGroup>
                    <RNSVGRect
                      height="24"
                      opacity={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M17 8h-1V6.11a4 4 0 1 0-8 0V8H7a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3v-8a3 3 0 0 0-3-3zm-7-1.89A2.06 2.06 0 0 1 12 4a2.06 2.06 0 0 1 2 2.11V8h-4zM18 19a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1v-8a1 1 0 0 1 1-1h10a1 1 0 0 1 1 1z"
                    />
                    <RNSVGPath
                      d="M12 12a3 3 0 1 0 3 3 3 3 0 0 0-3-3zm0 4a1 1 0 1 1 1-1 1 1 0 0 1-1 1z"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Password"
            placeholderTextColor="#8F9BB3"
            secureTextEntry={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                undefined,
              ]
            }
            value=""
          />
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              accessible={true}
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#8F9BB3"
              focusable={true}
              meetOrSlice={0}
              minX={0}
              minY={0}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              responsible={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "marginHorizontal": 8,
                    "tintColor": "#8F9BB3",
                    "width": 24,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4287601587}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup>
                  <RNSVGGroup>
                    <RNSVGRect
                      height="24"
                      opacity={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGCircle
                      cx="12"
                      cy="12"
                      r="1.5"
                    />
                    <RNSVGPath
                      d="M15.29 18.12L14 16.78l-.07-.07-1.27-1.27a4.07 4.07 0 0 1-.61.06A3.5 3.5 0 0 1 8.5 12a4.07 4.07 0 0 1 .06-.61l-2-2L5 7.87a15.89 15.89 0 0 0-2.87 3.63 1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25a9.48 9.48 0 0 0 3.23-.67z"
                    />
                    <RNSVGPath
                      d="M8.59 5.76l2.8 2.8A4.07 4.07 0 0 1 12 8.5a3.5 3.5 0 0 1 3.5 3.5 4.07 4.07 0 0 1-.06.61l2.68 2.68.84.84a15.89 15.89 0 0 0 2.91-3.63 1 1 0 0 0 0-1c-.64-1.11-4.16-6.68-10.14-6.5a9.48 9.48 0 0 0-3.23.67z"
                    />
                    <RNSVGPath
                      d="M20.71 19.29L19.41 18l-2-2-9.52-9.53L6.42 5 4.71 3.29a1 1 0 0 0-1.42 1.42L5.53 7l1.75 1.7 7.31 7.3.07.07L16 17.41l.59.59 2.7 2.71a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginVertical": 10,
            "opacity": 1,
          }
        }
        testID="confirmPasswordInput"
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#8F9BB3"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "marginHorizontal": 8,
                    "tintColor": "#8F9BB3",
                    "width": 24,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4287601587}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup>
                  <RNSVGGroup>
                    <RNSVGRect
                      height="24"
                      opacity={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M17 8h-1V6.11a4 4 0 1 0-8 0V8H7a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3v-8a3 3 0 0 0-3-3zm-7-1.89A2.06 2.06 0 0 1 12 4a2.06 2.06 0 0 1 2 2.11V8h-4zM18 19a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1v-8a1 1 0 0 1 1-1h10a1 1 0 0 1 1 1z"
                    />
                    <RNSVGPath
                      d="M12 12a3 3 0 1 0 3 3 3 3 0 0 0-3-3zm0 4a1 1 0 1 1 1-1 1 1 0 0 1-1 1z"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Re-enter Password"
            placeholderTextColor="#8F9BB3"
            secureTextEntry={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                undefined,
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3366FF",
            "borderColor": "#3366FF",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 10,
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
              },
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
